thunder/core/symbol.py

```diff
@@ -526,9 +527,16 @@ def __eq__(self, other):
         return (self.sym, self._var_args, self._var_output) == (other.sym, other._var_args, other._var_output)
 
     def rhs(self):
-        return BoundSymbolRHS(
-            self.sym, tuple(tree_flatten(self._var_args)[0]), tuple(tree_flatten(self._var_kwargs)[0])
-        )
+        def make_hashable(x: Any) -> tuple | FrozenDict:
+            if isinstance(x, dict):
+                return FrozenDict(x)
+            if isinstance(x, list):
+                return tuple(x)
+            return x
+
+        hashable_args = tuple(map(make_hashable, self._var_args))
```

Comment:
What if there's an argument like `[[0, 0], [1,]]`? Nested sequences are used for example in `prims.pad`: https://github.com/Lightning-AI/lightning-thunder/blob/6bc5752373c085a284413d84f420371212a9b996/thunder/core/prims.py#L3051
One possible solution is to make it recursive.

Reply:
On the latest commit it's already recursive and the snippet:
```python
tuple(map(make_hashable, [[0, 0], [1,]]))
```
generates `((0, 0), (1,))` which, being a tuple of tuples, hashable.

Maybe the code should be more explicit but in general, `make_hashable` recurses inside the lists here:
https://github.com/Lightning-AI/lightning-thunder/blob/e409cec8d03c7e8b7de9ada451501054da7d6eee/thunder/core/utils.py#L1130-L1131

Reply:
Thanks, GitHub probably showed me a different version for some reason.